### DISCUSSION TITLE
Callback forwarding

### DISCUSSION THREAD

**Discussion by dcpri** [slint-ui/slint#579] [Q&A] [answered]
I am having problems getting a global callback to work.
I tried following the sample code in Callback Forwarding Issue #111

In my individuals_add.rs file, I have

    let ui = AppWindow::new();

    ui.global::<Individual>().on_familyname_added({...

In my appwindow.60 file, I have the global

	export global Individual := {
    		callback familyname_added(string);
	}

In my ui_individuals.60 file, I have the LineEdit:

	edited(text) => {
		Individual.familyname_added(text);     
	}

But I get the error: 

  error: Cannot access id 'Individual'
    --> /Users/coding/genealogy/dcp-gen/ui/./ui_individuals.60:40:21

         	Individual.familyname_added(text); 

I tried declaring the global in all three files, without success.

**Comment by ogoffart** [comment, marked as answer]
You need to import it.

The best is to have the global in a "individual.60", so you can do 

```js
import { Individual } from "individual.60";
```

in every file that needs it.

And don't forget to re-export it from the appwindow.60: `export { Individual }`

**Reply by dcpri** [reply]
Thanks, again, for your help.
